virttest/qemu_devices/qcontainer.py

```diff
@@ -2935,22 +2934,45 @@ def dimm_device_define_by_params(self, params, name):
 
     def memory_define_by_params(self, params, name):
         """
-        Create memory modules by params, include memory object and
-        pc-dimm devices.
+        Create memory modules by params, include memory object,
+        pc-dimm, nvdimm and virtio-mem devices.
         """
         params = params.object_params(name)
         devices = []
         if not self.has_device("pc-dimm"):
             LOG.warn("'PC-DIMM' does not support by your qemu")
             return devices
+        mem_devtype = params.get("mem_devtype", "dimm")
         mem = self.memory_object_define_by_params(params, name)
         if mem:
             devices.append(mem)
             use_mem = params.object_params(name).get("use_mem", "yes")
             if use_mem == "yes":
-                dimm = self.dimm_device_define_by_params(params, name)
-                dimm.set_param("memdev", mem.get_qid())
-                devices.append(dimm)
+                if mem_devtype == "dimm":
+                    dimm = self.dimm_device_define_by_params(params, name)
+                    dimm.set_param("memdev", mem.get_qid())
+                    devices.append(dimm)
+                elif mem_devtype == "virtio":
```

Comment:
```suggestion
                elif mem_devtype == "virtio-mem":
```
since qemu has another virtio kind memdev - [virtio-pmem](https://qemu.readthedocs.io/en/latest/system/devices/virtio-pmem.html).

Reply:
Included now, thanks!